src/Temporalio.Extensions.Hosting/TemporalHostingServiceCollectionExtensions.cs

```diff
@@ -34,12 +34,13 @@ public static class TemporalHostingServiceCollectionExtensions
         /// the unique identifier for a worker in the service collection.
         /// </param>
         /// <returns>Options builder to configure the service.</returns>
+        [Obsolete("Use the AddHostedTemporalWorker overload with WorkerDeploymentOptions instead")]
         public static ITemporalWorkerServiceOptionsBuilder AddHostedTemporalWorker(
             this IServiceCollection services,
             string clientTargetHost,
             string clientNamespace,
             string taskQueue,
-            string? buildId = null) =>
+            string buildId) =>
```

Comment:
In order to make these obsolete, I had to make them not take null for the last param, and instead have that happen on the new one. I believe that should be a compatible change still

Reply:
With C# named parameters, it's technically not compatible since a user could have had `buildId: null`. Can you leave the `string?` type here (can still make the parameter required I believe).